src/background/adblocker.js

```diff
@@ -267,7 +267,18 @@ function injectScriptlets(filters, tabId, frameId, hostname) {
     }
 
     const func = scriptlet.func;
-    const args = parsed.args.map((arg) => decodeURIComponent(arg));
+    const scriptletGlobals = {
+      // Request a real extension resource to obtain a dynamic URL to the resource.
+      // Redirect resources are defined with `use_dynamic_url` restriction.
+      // refs https://developer.chrome.com/docs/extensions/reference/manifest/web-accessible-resources#manifest_declaration
+      warOrigin: chrome.runtime
```

Comment:
Can the detection of the base URL be performed once per service worker lifetime and saved for subsequent calls to `injectScriptlets`?

Reply:
I believe this can be cached:

> A dynamic ID is generated per session. That means it is regenerated when the browser restarts or the extension reloads.
>
> refs https://developer.chrome.com/docs/extensions/reference/manifest/web-accessible-resources